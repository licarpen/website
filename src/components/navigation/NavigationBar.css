.NavigationBar{
  display: flex;
  justify-content: flex-end;
  padding: 10px;
  margin-top: 10px;
  
  .NavLink{
    border: 3px white solid;
    color: white;
    border-radius: .6em;
    padding: 5px;
    padding-left: 10px;
    padding-right: 10px;
    margin-left: 10px;
    font-size: 30px;
    font-family: Arial, Helvetica, sans-serif;
    text-decoration: none;
  }
  .NavLink:hover{
    color: darkcyan;
    background: white;
  }
  .activeNavLink{
    color: darkcyan;
    background: white;
    outline: none;
  }

  @media screen and (max-width: 600px){
    justify-content: center;

    .NavLink{
      font-size: 20px;
    }
  }
}