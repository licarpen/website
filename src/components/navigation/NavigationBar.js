import React from 'react';
import { NavLink } from 'react-router-dom';
import styles from './NavigationBar.css';

const NavigationBar = () => {
  
  return (
    <div className={styles.NavigationBar}>
      <NavLink className={styles.NavLink} exact to='/'>
        Home
      </NavLink>
      <NavLink className={styles.NavLink} exact to='/about'>
        About
      </NavLink>
      <NavLink className={styles.NavLink} exact to='/connect'>
        Connect
      </NavLink>
    </div>
  );
};

export default NavigationBar;
