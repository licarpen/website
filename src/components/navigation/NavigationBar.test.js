import React from 'react';
import { shallow } from 'enzyme';
import NavigationBar from './NavigationBar';

describe('NavigationBar component', () => {
  it('matches a snapshot', () => {
    const navigationElements = [
      { label: 'Portfolio', value: 'portfolio' },
      { label: 'Connect', value: 'Connect' },
      { label: 'Home', value: 'home' }
    ];
    const wrapper = shallow(<NavigationBar navigationElements={navigationElements}name="sample name" onChange={() => {}}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
