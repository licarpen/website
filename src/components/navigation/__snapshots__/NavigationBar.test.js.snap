// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar component matches a snapshot 1`] = `
<Fragment>
  <NavigationElement
    key="portfolio"
    label="Portfolio"
    name="sample name"
    onChange={[Function]}
    value="portfolio"
  />
  <NavigationElement
    key="Connect"
    label="Connect"
    name="sample name"
    onChange={[Function]}
    value="Connect"
  />
  <NavigationElement
    key="home"
    label="Home"
    name="sample name"
    onChange={[Function]}
    value="home"
  />
</Fragment>
`;
